lib/iris/coords.py

```diff
@@ -2775,6 +2779,366 @@ def xml_element(self, doc):
+                f"to describe '{self.cf_role}' ."
+            )
+
+    def switch_start_index(self):
+        # Don't use a normal property setter - would be inappropriate since
+        # other properties are also modified.
+        """
+        Change :attr:`start_index` to its alternative value and change the
+        :attr:`indices` values to match. (:attr:`start_index` can be either
+        ``0`` or ``1``).
+
+        """
+        new_index = 1 - self.start_index
+        index_change = new_index - self.start_index
+        self._metadata_manager.start_index = new_index
+        # Modify the indices values to correspond with the new start_index.
+        self._values += index_change
+
+    def switch_element_dim(self):
+        # Don't use a normal property setter - would be inappropriate since
+        # other properties are also modified.
+        """
+        Change :attr:`element_dim` to its alternative value and restructure
+        :attr:`indices` to match. (:attr:`element_dim` can be ``0`` or ``1``).
+
+        """
+        self._metadata_manager.element_dim = 1 - self.element_dim
+        # Modify the indices structure to match the new element_dim.
+        # We know that indices is 2-dimensional so can just swap the two axes.
+        new_indices = self._values.swapaxes(0, 1)
+        # Replace the data manager with one for the new shaped indices.
+        # This is safe to override since Connectivity is not attached to a cube
+        # dimension unlike other _DimensionalMetadata-using classes.
+        self._values_dm = DataManager(new_indices)
+
+    def lazy_indices(self):
+        """
+        Return a lazy array representing the connectivity's indices.
+
+        Accessing this method will never cause the indices values to be loaded.
+        Similarly, calling methods on, or indexing, the returned Array
+        will not cause the connectivity to have loaded indices.
+
+        If the indices have already been loaded for the connectivity, the
+        returned Array will be a new lazy array wrapper.
+
+        Returns:
+            A lazy array, representing the connectivity indices array.
+
+        """
+        return super()._lazy_values()
+
+    def core_indices(self):
+        """
+        The indices array at the core of this connectivity, which may be a
+        NumPy array or a Dask array.
+
+        Returns:
+            numpy.ndarray or numpy.ma.core.MaskedArray or dask.array.Array
+
+        """
+        return super()._core_values()
+
+    def has_lazy_indices(self):
+        """
+        Return a boolean indicating whether the connectivity's :attr:`indices`
+        array is a lazy Dask array or not.
+
+        Returns:
+            boolean
+
+        """
+        return super()._has_lazy_values()
+
+    def __str__(self):
+        result = repr(self)
+        return result
+
+    def __repr__(self):
+        cls = type(self).__name__
+        result = (
+            f"{cls}({self.indices}), cf_role="
```

Comment:
There is an extra `)` here that requires to be removed i.e.,
```python
f"{cls}({self.indices}, cf_role="
```

Reply:
The `__str__` and `__repr__` don't marry up with the metadata, which ultimately I think is confusing e.g.,
```python
>>> c = Connectivity(np.arange(12).reshape(3, 4), "face_node_connectivity", "long-name", "var-name", dict(one="world"), 0, 1)
>>> print(c)
Connectivity([[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]), cf_role='face_node_connectivity', start_index=0, element_dim=1, long_name='long-name', var_name='var-name', attributes={'one': 'world'})
>>> print(c.metadata)
ConnectivityMetadata(standard_name=None, long_name='long-name', var_name='var-name', units=Unit('1'), attributes={'one': 'world'}, cf_role='face_node_connectivity', start_index=0, element_dim=1)
```

Reply:
Yup, that's not something I thought about at all 😳

Reply:
@bjlittle don't know if this info is useful to you but this problem started when I had the blinkers on and was looking at [`CellMeasure.__repr__()`](https://github.com/SciTools/iris/blob/3ff15a636f59d4cc3858199dfaaf83990e935154/lib/iris/coords.py#L861-L872), without noticing no other class overrides [`_DimensionalMetadata.__repr__()`](https://github.com/SciTools/iris/blob/3ff15a636f59d4cc3858199dfaaf83990e935154/lib/iris/coords.py#L324-L342); perhaps `CellMeasure` ought to also be fixed in future, since it creates a similar divergence from `CellMeasureMetadata.__repr__()`?

Reply:
2808afe2ea513b3549cdf4173bb87a6f74fd5c4b